packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -542,197 +888,143 @@ const MessageButton: React.FC<{
         dialogueUsers {
           _id
           displayName
-          karma
         }
         topUsers {
           _id
           displayName
           total_power
           total_agreement
+          recently_active_matchmaking
         }
        }
     }
   `);
 
-  const userDialogueUsefulData:UserDialogueUsefulData = data?.GetUserDialogueUsefulData
-
-  const {loading: userLoading, results: userDialogueChecks} = useMulti({
-    terms: {
-      view: "userDialogueChecks",
-      userId: currentUser?._id,
-      limit: 1000,
-    },
-    fragmentName: "DialogueCheckInfo",
-    collectionName: "DialogueChecks",
-  });
-
-  const {loading: userOptedInLoading, results: usersOptedInToDialogueFacilitation, loadMoreProps} = useMulti({
-    terms: { 
-      view: 'usersWithOptedInToDialogueFacilitation',
-      limit: 10, 
-        },
-    fragmentName: 'UsersOptedInToDialogueFacilitation',
-    collectionName: 'Users'  
-  });
-
   if (loading) {
     return <Loading />;
   } else if (!usersOptedInToDialogueFacilitation) {
     return <p>Error...</p>;
   }
 
-  const targetUserIds = userDialogueChecks?.map(check => check.targetUserId) ?? [];
-
-  async function createDialogue(title: string, participants: string[]) {
-    const createResult = await createPost({
-      data: {
-        title,
-        draft: true,
-        collabEditorDialogue: true,
-        coauthorStatuses: participants.map(userId => ({userId, confirmed: true, requested: false})),
-        shareWithUsers: participants,
-        sharingSettings: {
-          anyoneWithLinkCan: "none",
-          explicitlySharedUsersCan: "edit",
-        },
-        contents: {
-          originalContents: {
-            type: "ckEditorMarkup",
-            data: ""
-          }
-        } as AnyBecauseHard
-      },
-    });
-    if (createResult?.data?.createPost?.data) {
-      const post = createResult?.data?.createPost?.data;
-      if (post) {
-        const postId = post._id;
-        const postEditUrl = `/editPost?postId=${postId}`;
-        history.push(postEditUrl);
-      }
-    }
-  }
+  const userDialogueUsefulData: UserDialogueUsefulData = data?.GetUserDialogueUsefulData;
 
+  const matchedUsers: UsersOptedInToDialogueFacilitation[] | undefined = matchedUsersResult?.GetDialogueMatchedUsers;
+  const matchedUserIds = matchedUsers?.map(user => user._id) || [];
+  const topUsers = userDialogueUsefulData?.topUsers.filter(user => !matchedUserIds.includes(user._id));
+  const dialogueUsers = userDialogueUsefulData?.dialogueUsers.filter(user => !matchedUserIds.includes(user._id));
+  const optedInUsers = usersOptedInToDialogueFacilitation.filter(user => !matchedUserIds.includes(user._id));
+  
   if (!currentUser) return <p>You have to be logged in to view this page</p>
   if (loading) return <Loading />
   if (error || !userDialogueChecks || userDialogueChecks.length > 1000) return <p>Error </p>; // if the user has clicked that much stuff things might break...... 
```

Comment:
Why can't you just load less? This seems like the wrong solution

Reply:
We'll either have to filter out matched users on the client, or in the resolver on the server. As long this solution is about equally performant, seems fine to me?

Reply:
> equally performant

If you do the limit in the DB query, it's much more performant. (In practice I think this is a non-issue and you just shouldn't worry about this, unless you have a specific reason to expect performance issues.)